scripts/linux/github/updateLatestVersion.sh

```diff
@@ -0,0 +1,231 @@
+#    checked out.
+#######################################
+update_latest_version_json()
+{
+    check_required_variables
+
+    [[ -z "$AZURE_IOTEDGE_REPO_PATH" ]] && { echo "\$IOTEDGE_REPO_PATH is undefined"; exit 1; }
+
+    branchVersion=$(echo ${BRANCH_NAME##*/})
+    latestReleasedVersion=$(get_latest_release_per_branch_name)
+
+    if [ "$BRANCH_NAME" == "refs/heads/release/1.1" ]; then
+        # Set target version file to be updated
+        TARGET_IE_FILE="$AZURE_IOTEDGE_REPO_PATH/latest-iotedge-lts.json"
+
+        # Get all the relevant version to be verified
+        proposedEdgeletVersion=$(cat $IOTEDGE_REPO_PATH/edgelet/version.txt)
+        proposedImageVersion=$(cat $IOTEDGE_REPO_PATH/versionInfo.json | jq ".version" | tr -d '"')
+
+        content=$(cat $AZURE_IOTEDGE_REPO_PATH/latest-iotedge-lts.json)
+        latestEdgeletVersion=$(echo $content | jq ".iotedged" | tr -d '"')
+        latestImageVersion=$(echo $content | jq '."azureiotedge-agent"' | tr -d '"')
+
+        # Verify
+        echo "Sanity check iotedge version"
+        version_sanity_check $proposedEdgeletVersion $latestEdgeletVersion
+        echo "Sanity check docker image version"
+        version_sanity_check $proposedImageVersion $latestImageVersion
+
+        # Rewriting the latest-iotedge-lts.json
+        jqQuery=".iotedged = \"$proposedEdgeletVersion\" | .\"azureiotedge-agent\" = \"$proposedImageVersion\" | .\"azureiotedge-hub\" = \"$proposedImageVersion\""
+        echo $content | jq "$jqQuery" > $TARGET_IE_FILE
+
+        # Pring log for debugging
+        echo "Update $TARGET_IE_FILE:"
+        cat $TARGET_IE_FILE | jq '.'
+
+    elif [ "$BRANCH_NAME" == "refs/heads/release/1.2" ]; then
+
+        [[ -z "$IIS_REPO_PATH" ]] && { echo "\$IIS_REPO_PATH is undefined"; exit 1; }
+        # Set target version file to be updated
+        TARGET_IE_FILE="$AZURE_IOTEDGE_REPO_PATH/latest-aziot-edge.json"
+        TARGET_IIS_FILE="$AZURE_IOTEDGE_REPO_PATH/latest-aziot-identity-service.json"
+
+        # Get all the relevant version to be verified
+        proposedEdgeletVersion=$(cat $IOTEDGE_REPO_PATH/edgelet/version.txt)
+        proposedImageVersion=$(cat $IOTEDGE_REPO_PATH/versionInfo.json | jq ".version" | tr -d '"')
+        proposedIisVersion=$(grep "PACKAGE_VERSION:" $IIS_REPO_PATH/.github/workflows/packages.yaml | awk '{print $2}' | tr -d "'" | tr -d '"')
+
+        contentIe=$(cat $TARGET_IE_FILE)
+        latestEdgeletVersion=$(echo $contentIe | jq '."aziot-edge"' | tr -d '"')
+        contentIis=$(cat $TARGET_IIS_FILE)
+        latestIisVersion=$(echo $contentIis | jq '."aziot-identity-service"' | tr -d '"')
+
+        # Verify
+        echo "Sanity check iotedge version"
+        version_sanity_check $proposedEdgeletVersion $latestEdgeletVersion
+        echo "Sanity check docker image version"
+        version_sanity_check $proposedImageVersion get_latest_release_per_branch_name
```

Comment:
We're doing the sanity check for the Docker image version here. But We don't write the version anywhere (`latest-aziot-edge.json` only tracks the edgelet version). Do you know what that is?

Reply:
I'm not sure. @Gordonwang0 any thoughts?

Reply:
No, I don't know why we do this.

Reply:
My wildest recollection was that it has to do with Diagnostic module and how it basically uses aziot version to determine the edgeAgent & edgeHub version. But I'm not super familiar with that portion of code to take a look.

Reply:
@damonbarry After talking with Varun, it seems that we currently don't have code that checks if the latest EA/EH version are running.